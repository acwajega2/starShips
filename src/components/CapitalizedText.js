import React, { Component } from 'react';

class CapitalizedText extends Component {
  
    render() {
        const jsUcfirst = (string) => {
            return string.charAt(0).toUpperCase() + string.slice(1);
             
         }
    
        return (
            jsUcfirst(this.props.text)
            
        );
    }
}

export default CapitalizedText;