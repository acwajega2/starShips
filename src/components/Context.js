import React, { Component } from 'react';
import axios from 'axios' //--for perforning http requests

import Data from './Data'





//----Creating a starShip Context
const StarShipContext = React.createContext();

/*
1.We shall be using the StarShipContext context API to pass data across all the components using th StarShip Provider
2. We shall wrap the StarShip provider on top of our application

*/ 


//----> Creating the StartShipProvider

class StarShipProvider extends Component{

//----Creating/declaring the state
constructor (props){
    super(props)
    this.state ={
        starShips : [],
        err_msg : "",
        //----pagination variables
        pageNumbers: [],
        currentPage: 1,
        itemsPerPage: 3,
        totalItems : 0,
        totalPages : 0,
        hasLogedIn : false,
        currentUser: {},
        nextPage_Url: "",
        prevPage_Url: ""
        
    }
}
//----End of state declaration

componentDidMount(){
    /* */

    //----> Getting thr starShips on Component Mounting
    this.getStarShips()
    //this.initialiseStarShips()

    //IntialiseStarShips
}


componentDidUpdate(){
    /* */
}


componentWillUnmount(){
/* */
}


//  >>>>>>>>>>>>>>>>>>>   Implementing Functions to be used
initialiseStarShips = () =>{
    this.setState(() =>{
        return {
            starShips : Data.response,
            totalItems: Data.response.length,
            totalPages : Math.floor(Data.response.length / this.state.itemsPerPage)
            
        }
    })
    
  }


   
    
    

//-------> Get StarShips function
getStarShips = () =>{
    //----> we shall use axios to perform a get request and it will the return a promise
    axios.get("https://swapi.co/api/starships/")
    .then(res => {
        console.log(res.data.results)
        //----> Setting the starShip property of the starte
        this.setState(() =>{
            return {
                starShips: res.data.results,
                nextPage_Url: res.data.next,
                prevPage_Url:res.data.previous,
                totalItems: res.data.results.length,
                itemsPerPage: res.data.results.length,
                totalPages : Math.floor(res.data.results.length / this.state.itemsPerPage)
            }
        })

    })
}

    
//---------Function to navigate to the next webpage
    
    nextPage_link = () => {

        if (this.state.nextPage_Url !== null) {
            axios.get(this.state.nextPage_Url)
            .then(res => {
                console.log(res.data.results)
                //----> Setting the starShip property of the starte
                this.setState(() =>{
                    return {
                        starShips: res.data.results,
                        nextPage_Url: res.data.next,
                        prevPage_Url: res.data.previous,
                        itemsPerPage: res.data.results.length,
                        totalItems: res.data.results.length,
                        totalPages : Math.floor(res.data.results.length / this.state.itemsPerPage)
                    }
                })
        
            })

        }
    
}

    
//-------------------End of Next webpage

    
//-----------------Navigating to the previous page
    prevPage_link = () => {
        if (this.state.prevPage_Url !== null) {
            axios.get(this.state.prevPage_Url)
            .then(res => {
                console.log(res.data.results)
                //----> Setting the starShip property of the starte
                this.setState(() =>{
                    return {
                        starShips: res.data.results,
                        nextPage_Url: res.data.next,
                        prevPage_Url:res.data.previous,
                        totalItems: res.data.results.length,
                        itemsPerPage: res.data.results.length,
                        totalPages : Math.floor(res.data.results.length / this.state.itemsPerPage)
                    }
                })
        
            })

        }
        
    }
//--------------------End of Previous Page
showMessage =() =>{
    console.log("someone clicked changePage")

}

//------> Handle Social Login Success
handleSocialLogin =(user) => {

    console.log(user.profile)

    this.setState(()=>{
        return{
            hasLogedIn: true,
            user : user.profile,
            currentUser : user.profile

        }

    })

}


//--------



//---------> Handle Social Login Failure
handleSocialLoginFailure =(err) =>{
    this.setState(()=>{
        return{
            hasLogedIn: false,
            user : []

        }

    })

}



//--------


//-------->  Changing a page-----
changePage =(page) =>{       
    this.setState( () =>{
        return {
            currentPage: parseInt(page),
            err_msg : "sdsdsd",
        }
    }
   
    )
  }
//------- End of Changing page

//------> NextPage 
nextPage = () =>{
       this.state.currentPage !== this.state.totalPages + 1?
       this.setState( () =>{      
        return {
           currentPage : Number(this.state.currentPage) + 1,          
            err_msg : "sdsdsd",
        }
    })
        : 
        this.setState( () =>{      
            return {
               currentPage : Number(this.state.currentPage) ,              
                err_msg : "sdsdsd",
            }
        })
 }



//------> Previous Page
prevPage = () =>{
    this.state.currentPage !== 1?
    this.setState( () =>{   
     return {
        currentPage : Number(this.state.currentPage) - 1,       
         err_msg : "sdsdsd",
     }
 })
     : 
     this.setState( () =>{   
         return {
            currentPage : Number(this.state.currentPage) ,           
             err_msg : "sdsdsd",
         }
     })

}


//-----> Go to Home Page -----
homePage = () =>{
    this.setState( () =>{   
        return {
           currentPage : 1 ,           
            err_msg : "sdsdsd",
        }
    })

}
// >>>>>>>>> End of Functions

render(){

    return(
        <StarShipContext.Provider value={{...this.state, 
        getStarShips : this.getStarShips,
        changePage :this.changePage,
        showMessage :this.showMessage,
        nextPage : this.nextPage,
        prevPage : this.prevPage,
        homePage : this.homePage,    
        handleSocialLogin : this.handleSocialLogin,
            handleSocialLoginFailure: this.handleSocialLoginFailure,
            prevPage_link: this.prevPage_link,
            nextPage_link : this.nextPage_link
        
        }}>
             {this.props.children}
        </StarShipContext.Provider>
    )
}




}



//----->  Creating the Startship Consumer
const StarShipConsumer = StarShipContext.Consumer;


//------> So we shall export both the starship Provider and Consumer
export {StarShipProvider,StarShipConsumer}