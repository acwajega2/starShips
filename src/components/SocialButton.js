import React, { Component } from 'react';
import SocialLogin from 'react-social-login';
import { Avatar, Icon } from 'antd';

class SocialButton extends Component {
    render() {
        return (
            <Avatar icon="user"  onClick={this.props.triggerLogin} {...this.props}>
                { this.props.children }
            </Avatar>
           
        );
    }
}

export default SocialLogin(SocialButton);

