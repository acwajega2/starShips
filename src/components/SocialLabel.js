import React, { Component } from 'react';
import SocialLogin from 'react-social-login';
import { Menu  } from 'antd';

class SocialLabel extends Component {
    render() {
        return (
           <Menu.Item onClick={this.props.triggerLogin} {...this.props}>
               { this.props.children }
           </Menu.Item>
        );
    }
}

export default SocialLogin(SocialLabel);