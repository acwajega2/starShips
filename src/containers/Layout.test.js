import React from 'react';
import { shallow, mount } from '../Enzyme';

import renderer from 'react-test-renderer';
import { StarShipProvider } from '../components/Context';

import CustomLayout from './Layout';


describe("Testing the Custom Layout", () => {
    
    //------Doing SnapShot Testing---------------------------
    it("Matches the previous snap shot", () => {
        const sampleLayout = () => {
            
            return (
                <StarShipProvider>
                    <CustomLayout />                   
                 
                </StarShipProvider>
                
            )
        }    


            const component = shallow(sampleLayout())
            expect(component).toMatchSnapshot()
        });
    
    //------- End of Snapshot Testing--------------------
    
    //----- Testing if the StarShip Provider Initializes Properly
    it("Should Initialize properly", () => {
        const sampleLayout = () => {
            
            return (
                <StarShipProvider>
                    <CustomLayout >
                     
                    </CustomLayout>

                </StarShipProvider>
                
            )
        }   

            shallow(sampleLayout())
        })
    
        
    //---------End of Initialization Testing
        
    
    
    
    })