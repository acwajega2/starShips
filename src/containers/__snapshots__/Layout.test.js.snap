exports[`Testing the Custom Layout Matches the previous snap shot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StarShipProvider>
    <CustomLayout />
</StarShipProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <CustomLayout />,
      "value": Object {
        "changePage": [Function],
        "currentPage": 1,
        "currentUser": Object {},
        "err_msg": "",
        "getStarShips": [Function],
        "handleSocialLogin": [Function],
        "handleSocialLoginFailure": [Function],
        "hasLogedIn": false,
        "homePage": [Function],
        "itemsPerPage": 3,
        "nextPage": [Function],
        "pageNumbers": Array [],
        "prevPage": [Function],
        "showMessage": [Function],
        "starShips": Array [],
        "totalItems": 0,
        "totalPages": 0,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": undefined,
        "_currentValue2": undefined,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <CustomLayout />,
        "value": Object {
          "changePage": [Function],
          "currentPage": 1,
          "currentUser": Object {},
          "err_msg": "",
          "getStarShips": [Function],
          "handleSocialLogin": [Function],
          "handleSocialLoginFailure": [Function],
          "hasLogedIn": false,
          "homePage": [Function],
          "itemsPerPage": 3,
          "nextPage": [Function],
          "pageNumbers": Array [],
          "prevPage": [Function],
          "showMessage": [Function],
          "starShips": Array [],
          "totalItems": 0,
          "totalPages": 0,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": undefined,
          "_currentValue2": undefined,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
